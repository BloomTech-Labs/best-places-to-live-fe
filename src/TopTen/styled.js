import React from 'react';
import {makeStyles, withStyles} from "@material-ui/core";
import MenuItem from "@material-ui/core/MenuItem";
import Menu from '@material-ui/core/Menu';

 export const StyledMenu = withStyles({
    paper: {
        border: '1px solid #d3d4d5',
    },
})(props => (
    <Menu
        elevation={0}
        getContentAnchorEl={null}
        anchorOrigin={{
            vertical: 'bottom',
            horizontal: 'center',
        }}
        transformOrigin={{
            vertical: 'top',
            horizontal: 'center',
        }}
        {...props}
    />
));

export const StyledMenuItem = withStyles(theme => ({
    root: {
        '&:focus': {
            backgroundColor: theme.palette.primary.main,
            '& .MuiListItemIcon-root, & .MuiListItemText-primary': {
                color: theme.palette.common.white,
            },
        },
    },
}))(MenuItem);

export const useStyles = makeStyles(theme => ({
    root: {
        flexGrow: 1,
    },
}));
