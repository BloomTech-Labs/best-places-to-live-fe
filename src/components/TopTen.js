import React, { useEffect } from "react";
import { connect } from "react-redux";
import CityCard from "./CityCard";
import { Container, Flex, Text } from "../styles/index";
import SearchByFactors from "./SearchByFactors";
import Error from "./Error";
import { fetchTopTen } from "../actions/topTen";

function TopTen({ displayedCities, fetchTopTenError, fetchTopTen }) {
  useEffect(() => fetchTopTen({}), []);

  return (
    <Container p={`0 15px`}>
      {fetchTopTenError && <Error error={fetchTopTenError} />}
      <Flex
        flexDirection="column"
        alignItems="center"
        display="flex"
        justifyContent="center"
      ></Flex>
      <Text as="h2" textAlign="center">
        {" "}
        Top Ten
      </Text>
      <Flex display="flex" justifyContent="center" flexWrap="wrap">
        {displayedCities.map(city => (
          <CityCard key={city._id} city={city} />
        ))}
      </Flex>
    </Container>
  );
}

const mapStatetoProps = state => {
  return {
    displayedCities: state.displayedCities,
    fetchTopTenError: state.fetchTopTenError
  };
};

export default connect(mapStatetoProps, { fetchTopTen })(TopTen);
