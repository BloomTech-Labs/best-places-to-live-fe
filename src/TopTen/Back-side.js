import React from 'react'
import costofliving from "../images/budget.png";
import commute from "../images/travel.png";
import safety from "../images/police-car.png";
import tolerance from "../images/tolerance.png";
import {TopTenCardBack,CardIcon,ScoreColumn,TotalScore} from "./styled";


const TopTenBack = props =>  {
    return(
        <>
            <TopTenCardBack>

                <ScoreColumn>
                    <CardIcon src={costofliving}/>
                    {Math.round(props.card.score_cost_of_living)}
                </ScoreColumn>

                <ScoreColumn>
                    <CardIcon src={commute}/>
                    {Math.round(props.card.score_commute)}
                </ScoreColumn>

                <ScoreColumn>
                    <CardIcon src={safety}/>
                    {Math.round(props.card.score_safety)}
                </ScoreColumn>

                <ScoreColumn>
                    <CardIcon src={tolerance}/>
                    {Math.round(props.card.score_tolerance)}
                </ScoreColumn>

                <TotalScore>
                  Total Score:{Math.round(props.card.score_total)}
                </TotalScore>

            </TopTenCardBack>
        </>
    );
};

export default TopTenBack;