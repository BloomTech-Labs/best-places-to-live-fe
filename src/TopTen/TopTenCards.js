import React from "react";
import {CardDetail, TopTenImg} from "./styled";

const TopTenCards = props => {
  return (
      <>

          <TopTenImg alt="Card-Image"src={props.card.photo} width="300px" height="200px" />
              <CardDetail>
                  <p>{props.card.name}</p>
          </CardDetail>
      </>

  );
};

export default TopTenCards;
