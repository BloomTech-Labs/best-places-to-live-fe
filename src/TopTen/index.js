import React, { useState, useEffect } from "react";
import axios from "axios";
import { TopTenContainer } from './styled';

import TopTenCards from "./TopTenCards";




const TopTen = () => {
  const [topTenList, settopTenList] = useState([]);



  useEffect(() => {
    const fetchTopTen = () => {
      axios
        .get("/city/topten-cost-of-living/")
        .then(res => {
          settopTenList(res.data.cities);
        })
        .catch(err => {
          console.log(err);
        });
    };
    fetchTopTen();
  }, []);


  return (
    <div>
      <div>
        <h1>Top Cities</h1>
        <button
          aria-controls="customized-menu"
          aria-haspopup="true"
          variant="contained"
          color="primary">
          Filter
        </button>
      </div>
      <TopTenContainer>
            {topTenList
              ? topTenList.map(state => (
                    <TopTenCards key={state.id} card={state} />
                ))
              : ""}
      </TopTenContainer>
    </div>
  );
};

export default TopTen;