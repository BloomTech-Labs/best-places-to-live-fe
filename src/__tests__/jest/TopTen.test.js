import React from "react";
import renderer from "react-test-renderer";
import TopTen from "../../TopTen";

describe("<TopTen />", () => {
  it("Matches Snapshot", () => {
    const tree = renderer.create(<TopTen />).toJSON();
    // @ts-ignore
    expect(tree).toMatchSnapshot();
  });
});
