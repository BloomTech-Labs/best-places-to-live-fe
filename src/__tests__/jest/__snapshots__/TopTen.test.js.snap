// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TopTen /> Matches Snapshot 1`] = `
<div
  className="sc-bdVaJa iHvHEW"
>
  <div
    className="sc-bwzfXH fJCSGU"
  >
    <h1>
      Let's move homie! Explore the places to call home.
    </h1>
    <p>
      Customize your search. Click on a category. Find a place you'll love.
    </p>
  </div>
  <div
    className="sc-ifAKCX jricq"
  >
    <img
      className="sc-EHOje cchfUa"
      onClick={[Function]}
      src="budget.png"
    />
    <img
      className="sc-EHOje cchfUa"
      onClick={[Function]}
      src="travel.png"
    />
    <img
      className="sc-EHOje cchfUa"
      onClick={[Function]}
      src="police-car.png"
    />
    <img
      className="sc-EHOje cchfUa"
      onClick={[Function]}
      src="tolerance.png"
    />
  </div>
  <div
    className="sc-htpNat ioYqPc"
  />
</div>
`;
